test/tests/selftest/pbs_pbstestsuite.py

```diff
@@ -0,0 +1,222 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2018 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.selftest import *
+from ptl.utils.pbs_snaputils import *
+
+
+class TestPBSTestSuite(TestSelf):
+    """
+    Contains tests for pbs_testsuite module's functionality
+    """
+
+    def test_revert_pbsconf_onehost(self):
+        """
+        Test the functionality of PBSTestSuite.revert_pbsconf()
+        for a single host type 1 installation
+        """
+        pbs_conf_filepath = self.du.get_pbs_conf_file(self.server.hostname)
+        pbs_conf_val = self.du.parse_pbs_config(self.server.hostname,
+                                                pbs_conf_filepath)
```

Comment:
I just checked, you don't need to call get_pbs_conf_file() before parse_pbs_conf.  If the second argument is not passed to parse_pbs_config(), it will default to calling get_pbs_conf_file(). (and other places in the code)

Reply:
I'm aware of that, the reason I'm doing it this way is so that it doesn't have to call get_pbs_conf_file() again when I set new pbs.conf attributes inside set_pbs_config()

Reply:
that's exactly why I was doing it, I guess it won't be too much of an overhead so I'll do what you suggest